fhir-search/src/test/java/com/ibm/fhir/search/test/BaseSearchTest.java

```diff
@@ -68,35 +77,27 @@ public void clearThreadLocal() {
         FHIRRequestContext.remove();
     }
 
-    /**
-     * This function returns a list containing the names of the SearchParameters contained in the input list.
-     *
-     * @param spList
-     *            the list of SearchParameter from which to collect the names
-     * @return the list of search parameter names
-     */
-    protected List<String> getSearchParameterCodes(List<SearchParameter> spList) {
-        List<String> result = new ArrayList<>();
-        for (SearchParameter sp : spList) {
-            result.add(sp.getCode().getValue());
-        }
-        return result;
-    }
-
     /*
      * The SearchParameters return an array of values, now the printSearchParameters returns all values.
      * @param label
      * @param spList
      */
-    protected void printSearchParameters(String label, List<SearchParameter> spList) {
-        if (DEBUG) {
+    @SuppressWarnings("unused")
+    protected void printSearchParameters(String label, Map<String, SearchParameter> parameters) {
+        if (DEBUG && parameters != null) {
             System.out.println("\nTest: " + label + "\nSearch Parameters:");
-            for (SearchParameter sp : spList) {
+            for (Entry<String, SearchParameter> entry : parameters.entrySet()) {
+                String code = entry.getKey();
+                SearchParameter sp = entry.getValue();
+
+                String url = sp.getUrl().getValue();
+                String version = (sp.getVersion() == null) ? null : sp.getVersion().getValue();
+                String canonical = (version == null) ? url : url + "|" + version;
+
                 List<ResourceType> resources = sp.getBase();
                 for (ResourceType resource : resources) {
-                    System.out.println("\t" + resource.getValue() + ":" + sp.getCode().getValue());
+                    System.out.println("\t" + resource.getValue() + ":" + code + ":" + canonical);
                 }
-
             }
         }
```

Comment:
Remove it then?

Reply:
the warning that we're suppressing here is the one that is caused by DEBUG being false.
eclipse was warning us that the whole method is basically dead code, but thats actually the intent...it only does something if DEBUG is flipped to true.